File: packages/sdk-python/veltrix_sdk/resources/feature_flags.py
# -*- coding: utf-8 -*-

from .base import BaseResource


class FeatureFlagsResource(BaseResource):
    """Public feature-flags endpoint (``/api/feature-flags``).

    Used by clients to conditionally enable UI/behaviour. Public — no
    authentication required.
    """
    RESOURCE_PATH = "feature-flags"

    def get(self, **kwargs):
        """Retrieves the deployment's feature flags (GET /api/feature-flags)."""
        return self._list(**kwargs)

File: packages/sdk-js/src/resources/feature_flags.ts
import { BaseResource } from './base';
import { AxiosRequestConfig } from 'axios';

// Public feature-flags endpoint (/api/feature-flags). Used by clients to
// conditionally enable UI/behaviour. Public — no authentication required.

export class FeatureFlagsResource extends BaseResource {
  protected readonly RESOURCE_PATH = 'feature-flags';

  /** Retrieves the deployment's feature flags. GET /api/feature-flags */
  async get(config?: AxiosRequestConfig): Promise<Record<string, any>> {
    return this._list<Record<string, any>>(undefined, config);
  }
}
